Fix failing build (due to Pandas 1.4.0)

The failure is coming from statsmodels:

```
ERROR collecting sgkit/tests/test_association.py ___________________________________________________________________
../../miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/statsmodels/compat/pandas.py:61: in <module>
    from pandas import NumericIndex
E   ImportError: cannot import name 'NumericIndex' from 'pandas' (/Users/tom/miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/pandas/__init__.py)

During handling of the above exception, another exception occurred:
sgkit/tests/test_association.py:37: in <module>
    import statsmodels.api as sm
../../miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/statsmodels/api.py:71: in <module>
    from . import datasets, distributions, iolib, regression, robust, tools
../../miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/statsmodels/distributions/__init__.py:4: in <module>
    from .discrete import genpoisson_p, zipoisson, zigenpoisson, zinegbin
../../miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/statsmodels/distributions/discrete.py:6: in <module>
    from statsmodels.base.model import GenericLikelihoodModel
../../miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/statsmodels/base/model.py:9: in <module>
    from statsmodels.base.data import handle_data
../../miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/statsmodels/base/data.py:13: in <module>
    from statsmodels.tools.decorators import cache_readonly, cache_writable
../../miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/statsmodels/tools/decorators.py:2: in <module>
    from statsmodels.compat.pandas import cache_readonly as PandasCacheReadonly
../../miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: in <module>
    from pandas import Int64Index as NumericIndex
<frozen importlib._bootstrap>:1039: in _handle_fromlist
    ???
../../miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/pandas/__init__.py:198: in __getattr__
    warnings.warn(
E   FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
```

It looks like this is fixed in https://github.com/statsmodels/statsmodels/issues/7683, but there is no release yet.

I tried running `pytest -W ignore::FutureWarning`, but then I get lots of bgen test failures like

```
sgkit/tests/io/bgen/test_bgen_reader.py::test_read_bgen[chunks0] - ValueError: Metadata mismatch found in `from_delayed`
```

I haven't looked into that any further. Perhaps we should just restrict to `pandas < 1.4.0`?

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,7 +1,7 @@
 numpy < 1.22
 xarray
-dask[array] >= 2021.10.0
-distributed >= 2021.10.0
+dask[array] >= 2021.10.0, <= 2022.01.0
+distributed >= 2021.10.0, <= 2022.01.0
 dask-ml
 scipy
 typing-extensions

--- a/requirements.txt
+++ b/requirements.txt
@@ -9,3 +9,4 @@ numba
 zarr
 fsspec != 2021.6.*
 scikit-learn
+pandas < 1.4.0

--- a/setup.cfg
+++ b/setup.cfg
@@ -28,8 +28,8 @@ python_requires = >=3.7
 install_requires =
     numpy < 1.22
     xarray
-    dask[array] >= 2021.10.0
-    distributed >= 2021.10.0
+    dask[array] >= 2021.10.0, <= 2022.01.0
+    distributed >= 2021.10.0, <= 2022.01.0
     dask-ml
     scipy
     zarr
@@ -37,6 +37,7 @@ install_requires =
     typing-extensions
     fsspec != 2021.6.*
     scikit-learn
+    pandas < 1.4.0
     setuptools >= 41.2  # For pkg_resources
 setup_requires =
     setuptools >= 41.2

